sechub-analyzer-cli/src/main/java/com/daimler/sechub/analyzer/core/Processor.java

```diff
@@ -0,0 +1,117 @@
+package com.daimler.sechub.analyzer.core;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.daimler.analyzer.model.AnalyzerResult;
+import com.daimler.analyzer.model.MarkerPair;
+
+public class Processor {
+    private final static Logger logger = LogManager.getLogger(Processor.class.getName());
+    
+    public Processor() {}
+    
+    /**
+     * Processes the given files
+     * 
+     * @param file process a file or folder
+     * @return
+     * @throws FileNotFoundException
+     */
+    public AnalyzerResult processFiles(List<String> filePaths) throws FileNotFoundException {
+        Map<String, List<MarkerPair>> result = new HashMap<>();
+        List<File> files = new LinkedList<>();
+        
+        for (String filePath : filePaths) {
+            
+            File file = new File(filePath);
+    
+            if (file.exists()) {
+                files.add(file);
+            } else {
+                throw new FileNotFoundException("File not found: " + filePath);
+            }
+        }
+        
+        result = analyzeFiles(files);
+        
+        AnalyzerResult analyzerResult = new AnalyzerResult(result);
+        
+        return analyzerResult;
+    }
+
+    /**
+     * Analyze files
+     * 
+     * Creates a list of all files which have be be analyzed.
+     * Starts the file analysis of all files in the previously created list.
+     * 
+     * @param rootFiles
+     * @return all files with markers
+     */
+    protected Map<String, List<MarkerPair>> analyzeFiles(List<File> rootFiles) {
+        Map<String, List<MarkerPair>> result = new HashMap<>();
+
+        Set<File> files = new HashSet<>();
+        
+        // create a list of all files
```

Comment:
I prefer block comments for comments - most times I use //  either for parts to remove, or when being a "sub comment"